capstone/src/main/java/com/google/sps/servlets/UpdateAssociationServlet.java

```diff
@@ -37,13 +54,21 @@ public void init() throws ServletException {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    AssociationAnalysis association = new AssociationAnalysis(loadPreviousResults());
     CloudNLPAssociation nlp = new CloudNLPAssociation(nlpClient);
-    ArrayList<AssociationResult> res =
-        association.calculateScores(nlp.analyzeAssociations(getComments()));
-    response.getWriter().println(res);
 
-    storeResults(res);
+    ArrayList<EntitySentiment> sentiments = nlp.analyzeAssociations(getComments());
+
+    for (String scope : SCOPES) {
+      AssociationAnalysis analysis = new AssociationAnalysis(loadPreviousResults(scope));
```

Comment:
As you add more and more comments to the datastore, I worry that this will become more and more time consuming.

As a future request, I recommend you guys add a time filter so that you don't get all comments but just the most recent ones!

Reply:
I currently only store the comments that haven't been processed yet. I would like to filter within datastore to reduce overhead in request, but its currently giving index errors due to some structure issue (I'll take a look at this in the next week).